lib/mongo/session.ex

```diff
@@ -339,6 +339,7 @@ defmodule Mongo.Session do
   @impl :gen_statem
   # Abort all pending transactions if there any and end session itself.
   def terminate(_reason, state, %{pid: pid} = data) do
+    _=
```

Comment:
Is this needed?

Reply:
I think the compiler was complaining.  Also, I think sticking to functional coding is good.  Any  nesting construct should return value and shall be assigned to something.

Reply:
Personally I often just leave it as is, without assignment unless the Dialyzer or compiler issue warnings.

Reply:
It was dialyzer (not compiler).  It had following iddue when I didn't have  _=. Adding that fixed the issue ;

**lib/mongo/session.ex:343:unmatched_return
The expression produces a value of type:

nil | :ok | {:error, %Mongo.Error{_ => _}}

but this value is unmatched.**